### PR TITLE
Exploit Patch Documentation

### PR DESCRIPTION
**Pull request by Flibio**
Documents how Sponge patches various exploits in Vanilla Minecraft. Closes issue #402.

Depends on #413 to update the global.conf page

---

[Main Preview](http://flibio.net/preview/exploit-patches/)

[File Changed](http://flibio.net/preview/exploit-patches/server/management/exploit-patches/)

### CODE REVIEW

File: source/server/management/exploit-patches.rst [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+Exploits Patched by Sponge
+==========================
+
+#. Sign command exploit where a client could run a command such as 'op'
+#. Client could force the server to make the user respawn invisible
+#. Client could set an itemstack's display name and cause it to exceed the character limit
+
+.. warning::
+    The invisibility exploit has been disabled in recent Sponge builds due to false triggering.
```
Comment (saladoc) [inline source/server/management/exploit-patches.rst]: so the exploit cannot be used? or the fix for the exploit has been disabled? Also please clarify the "false triggering"

Comment (Tzky) [inline source/server/management/exploit-patches.rst]: My assumtion is that the exploit patch had too many false positives (players getting called cheater by the patch for no reason), thus the patch was temporarily disabled,

**Comment by Tzky** [comment]
I suggest explaining where (in the config) you have to enable/disable the exploit patches. A link to the server-properties page might also suffice

**Comment by Flibio** [comment]
@Tzky Maybe a link to the global.conf page instead?

**Comment by Tzky** [comment]
Besides the minor typo, this looks good.

**Comment by Inscrutable** [comment]
One problem I can see: It uses *Sponge* as shorthand for the *implementations*. This needs to be understood (or corrected). The patches should apply to both SpongeForge and SpongeVanilla, but Sponge usually refers to the entire project *per-se*.

**Comment by Flibio** [comment]
@Inscrutable I updated the PR based on your comment. Let me know what you think.

**Comment by Tzky** [comment]
Another suggestion: Please don't squash directly, instead wait until the PR is fully done. It's easier for us to spot the changes this way :)

File: source/server/management/exploit-patches.rst [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+In recent Sponge builds (SpongeForge 974+), SpongeForge and SpongeVanilla patch a few client-server exploits. Whenever
+the implementations detect a user performing an exploit, they are kicked from the server with a message explaining why
+they were kicked. If enabled, a log message is also sent to the console. More exploit patches may be added in the
+future.
+
+Exploits Patched by Sponge Implementations
+==========================================
+
+#. Sign command exploit where a client could run a command such as 'op'
```
Comment (kenzierocks) [inline source/server/management/exploit-patches.rst]: Might want to include the conf entry here (and below), like `logging.exploit-sign-command-updates`, for quick and easy reference.

**Comment by Flibio** [comment]
@kenzierocks I was planning on that, but instead left a link to the global.conf page where it was explained in more detail.

**Comment by kenzierocks** [comment]
Ok, I guess that works. To me, if I was looking to disable it I would prefer if the keys were available immediately instead of having to scan the `global.conf` page.

**Comment by Tzky** [comment]
I think showing the config options off on this page as well as linking to the global.conf page is fine.

**Comment by Tzky** [comment]
@progwml6 suggested that we add a short notice *where* users should report exploits if they know of any. This will help us track them down and providing a fix/logs by default.

Suggested places to report:
* PM on forums
* PM on IRC
* better **not** post publicly on the forums if it's a still unknown issue

**Comment by progwml6** [comment]
i would not reccomend IRC pm's as those can easily get lost/ignored when staff are busy.
an email adress that forwards to our developers might be a good option for this

**Comment by Inscrutable** [comment]
This is probably a good place to include such a note, so that searching for 'exploit' takes you to this page. I'd be a little stronger with the wording too - "DO NOT post public messages anywhere detailing the exploit you found, that can make things worse. Don't even create a GitHub issue, notify us by PM or email, and keep it confidential until it's patched"

**Comment by Tzky** [comment]
closed via 45fdf94 and  aa55eda

**Pull request closed without merging**